create-quasar/templates/app/quasar-v2/ts-webpack/BASE/src/components/EssentialLink.vue

```diff
@@ -19,8 +19,17 @@
   </q-item>
 </template>
 
-<script lang="ts">
-<% if (typescriptConfig === 'composition') { %>import { defineComponent } from 'vue';
+<% if (typescriptConfig === 'composition-setup') { %><script setup lang="ts">
+export interface EssentialLinkProps {
+  title: string;
+  caption?: string;
+  link?: string;
+  icon?: string;
+}
+const props = withDefaults(defineProps<EssentialLinkProps>(), {
+  link: '#',
+});<% } else if (typescriptConfig === 'composition') { %><script lang="ts">
```

Comment:
See https://github.com/quasarframework/quasar/pull/13949#discussion_r924241462
```suggestion
withDefaults(defineProps<EssentialLinkProps>(), {
  caption: '',
  link: '#',
  icon: '',
});<% } else if (typescriptConfig === 'composition') { %><script lang="ts">
```